ee/hogai/tool.py

```diff
@@ -88,10 +88,13 @@ class MaxTool(BaseTool):
     _config: RunnableConfig
     _state: AssistantState
 
-    @abstractmethod
     def _run_impl(self, *args, **kwargs) -> tuple[str, Any]:
         """Tool execution, which should return a tuple of (content, artifact)"""
-        pass
+        raise NotImplementedError
+
+    async def _arun_impl(self, *args, **kwargs) -> tuple[str, Any]:
+        """Tool execution, which should return a tuple of (content, artifact)"""
+        raise NotImplementedError
```

Comment:
It's significant extra mental overhead to know which of these two `_run_impl`/`_arun_impl` methods to implement as a MaxTool author. Can we just decide on one version being canonical? We shouldn't force e.g. the experiments team to know much about async vs. sync variants of assistant execution